src/bytecode/instructions.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt::{Display, Formatter, Result};
+
+/// Represents all values that Lua supports
+/// # Notes
+/// Not all values are represented as the current implementation only supports
+/// a small subset of Lua
+#[derive(Clone)]
+pub enum Value {
+    // Lua specific
+    Nil,
+    Boolean(bool),
+    Number(f64),
+    Str(String),
+    // Implementation specific
+    // Variables are converted to other values when interpreted
+    Var(String)
+}
+
+impl Display for Value {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        match *self {
+            Value::Nil => write!(f, "Nil"),
+            Value::Boolean(b) => write!(f, "{}", b.to_string()),
+            Value::Number(float) => write!(f, "{}", float),
+            Value::Str(ref content) => write!(f, "\"{}\"", content),
+            Value::Var(ref name) => write!(f, "{}", name)
+        }
+    }
+}
+
+/// The instructions supported by the LuaBytecode
+pub enum Instr {
+    Mov(String, Value),
```

Comment:
I guess this is supposed to be the register instructions? In general -- and definitely here -- we don't want to use strings for anything that's performance critical: strings require heap allocation and comparisons are expensive. Although we don't have to do it in this PR, what we really want in the long run is a compact integer encoding of instructions. That almost certainly means that we won't directly be able to represent things as a Rust enum though.

Reply:
I wanted to have something that works (which I can later optimise). I totally agree that `String`s are quite heavy. I will try to substitute the string with an integer in a later PR, but first I wanted to have a dumb/slow interpreter that can interpret most of the Lua language.

Reply:
OK.